internal/monitors/docker/docker.go

```diff
@@ -225,9 +234,63 @@ func parseContainerEnvSlice(env []string) map[string]string {
 	return out
 }
 
+// EnableExtraGroups enables extra metrics that were individually turned on
+// by ExtraMetrics/ExtraGroups configuration
+func EnableExtraGroups(initConf EnhancedMetricsConfig, extraGroups, extraMetrics []string) EnhancedMetricsConfig {
+	groupEnableMap := map[string]bool{
+		groupBlkio: initConf.EnableExtraBlockIOMetrics,
+		groupCPU: initConf.EnableExtraCPUMetrics,
+		groupMemory: initConf.EnableExtraMemoryMetrics,
+		groupNetwork: initConf.EnableExtraNetworkMetrics,
+	}
+
+	for _, group := range extraGroups {
+		groupEnableMap[group] = true
+	}
+	for _, metric := range extraMetrics {
+		if metricSet[metric].Group != "" {
+			groupEnableMap[metricSet[metric].Group] = true
+		}
+	}
+
+	return EnhancedMetricsConfig{
+		EnableExtraBlockIOMetrics: groupEnableMap[groupBlkio],
+		EnableExtraCPUMetrics:     groupEnableMap[groupCPU],
+		EnableExtraMemoryMetrics:  groupEnableMap[groupMemory],
+		EnableExtraNetworkMetrics: groupEnableMap[groupNetwork],
+	}
+}
+
 // Shutdown stops the metric sync
 func (m *Monitor) Shutdown() {
 	if m.cancel != nil {
 		m.cancel()
 	}
+
+}
+
+// GetExtraMetrics returns additional metrics that should be allowed through.
+func (c *Config) GetExtraMetrics() []string {
+	var extraMetrics []string
+
+	if c.EnableExtraBlockIOMetrics {
+		extraMetrics = append(extraMetrics, groupMetricsMap[groupBlkio]...)
+	}
+
+	if c.EnableExtraCPUMetrics {
+		extraMetrics = append(extraMetrics, groupMetricsMap[groupCPU]...)
+	}
+
+	if c.EnableExtraMemoryMetrics {
+		extraMetrics = append(extraMetrics, groupMetricsMap[groupMemory]...)
+	}
+
+	if c.EnableExtraNetworkMetrics {
+		extraMetrics = append(extraMetrics, groupMetricsMap[groupNetwork]...)
+	}
+	
+	logger := logrus.WithFields(logrus.Fields{"monitorType": "GetExtraMetrics"})
+	logger.WithFields(logrus.Fields{"extraMetrics":extraMetrics}).Debug("Extra metrics")
```

Comment:
Not sure it should be logged per-monitor like this, I'll let @keitwb chime in.

Reply:
oops this is just a blip. was for debugging. getting rid of this